python/pyarrow/table.pxi

```diff
@@ -95,6 +95,29 @@ cdef class ChunkedArray:
             else:
                 index -= self.chunked_array.chunk(j).get().length()
 
+    def dictionary_encode(self):
+        """
+        Compute dictionary-encoded representation of array
```

Comment:
Can you list the expected return type (still chunked array, but the chunks will share a common dictionary)?